src/averageAges.js

```diff
@@ -37,7 +42,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const moms = people.map(person => person.mother);
+  const women = people.filter(person => person.sex === 'f');
+  const filteredWomen = withChildren
+    ? women.filter(person => moms.some(mom => mom === person.name))
+    : women;
```

Comment:
the same call filter and pass different callbacks

Reply:
i fixed all, but for me if that doesn't slow down code, my varian more readable

Reply:
except the first :)

Reply:
but it make code work slower)